Tests/Admin/Extension/LockExtensionTest.php

```diff
@@ -52,240 +51,179 @@ class LockExtensionTest extends \PHPUnit_Framework_TestCase
-        $this->admin->expects($this->any())
-            ->method('getUniqId')
-            ->will($this->returnValue($uniqid));
-
-        $this->configureAdminRequest();
+        $this->modelManager->lock()->shouldNotBeCalled();
 
         $this->request->request->set($uniqid, array('something'));
-
-        $this->modelManager->expects($this->never())
-            ->method('lock');
-
-        $this->lockExtension->preUpdate($this->admin, $this->object);
+        $this->lockExtension->preUpdate($this->admin->reveal(), $this->object);
     }
 
     public function testPreUpdateIfModelManagerIsNotImplementingLockerInterface()
     {
-        $modelManager = $this->getMock('Sonata\AdminBundle\Model\ModelManagerInterface');
+        $modelManager = $this->prophesize('Sonata\AdminBundle\Model\ModelManagerInterface');
         $uniqid = 'admin123';
-
-        $this->admin->expects($this->any())
-            ->method('getUniqId')
-            ->will($this->returnValue($uniqid));
-
-        $this->configureAdminRequest();
+        $this->configureAdmin($uniqid, $this->request, $modelManager->reveal());
 
         $this->request->request->set($uniqid, array('_lock_version' => 1));
-
-        $this->admin->expects($this->any())
-            ->method('getModelManager')
-            ->will($this->returnValue($modelManager));
-
-        $this->modelManager->expects($this->never())
-            ->method('lock');
-
-        $this->lockExtension->preUpdate($this->admin, $this->object);
+        $this->lockExtension->preUpdate($this->admin->reveal(), $this->object);
     }
 
     public function testPreUpdateIfObjectIsVersioned()
     {
         $uniqid = 'admin123';
+        $this->configureAdmin($uniqid, $this->request, $this->modelManager->reveal());
 
-        $this->modelManager->expects($this->once())
-            ->method('lock')
-            ->with($this->object, 1);
-
-        $this->configureAdminRequest();
+        $this->modelManager->lock($this->object, 1)
+            ->shouldBeCalled();
 
         $this->request->request->set($uniqid, array('_lock_version' => 1));
-
-        $this->admin->expects($this->any())
-            ->method('getModelManager')
-            ->will($this->returnValue($this->modelManager));
-
-        $this->admin->expects($this->any())
-            ->method('getUniqId')
-            ->will($this->returnValue($uniqid));
-
-        $this->lockExtension->preUpdate($this->admin, $this->object);
+        $this->lockExtension->preUpdate($this->admin->reveal(), $this->object);
     }
 
     private function configureForm()
     {
-        $form = $this->getMock('Symfony\Component\Form\FormInterface');
+        $form = $this->prophesize('Symfony\Component\Form\FormInterface');
+
+        $form->getData()
+            ->willReturn($this->object);
 
-        $form->expects($this->any())
-            ->method('getData')
-            ->will($this->returnValue($this->object));
+        $form->getParent()
+            ->willReturn(null);
```

Comment:
I though about doing in one line, but I wasn't sure. I think almost all the mocked methods can be written in one line. Should I rewrite all of them in one line?